docs/api/navigation-history.md

```diff
@@ -0,0 +1,26 @@
+## Class: NavigationHistory
+
+> Handles session history, maintaining a list of navigation entries for backward and forward navigation.
+Process: [Main](../glossary.md#main-process)<br />
+_This class is not exported from the `'electron'` module. It is only available as a return value of other methods in the Electron API._
+
+### Instance Methods
+
+#### `navigationHistory.getActiveIndex()`
+
+Returns `Integer` - The index from which we would go back/forward or reload.
+
+#### `navigationHistory.getEntryAtIndex(index)`
+
+* `index` Integer
+
+Returns `Object`:
+
+* `url` string - The url of the navigation entry at the given index.
+* `title` string - The app title set by the page to be displayed on the tab.
+
+If index is invalid (greater than history length or less than 0), null will be returned.
```

Comment:
```suggestion
If index is out of bounds (greater than history length or less than 0), null will be returned.
```

Reply:
This is also a dumb question but what happens when you pass in something like 1.5 on runtime as an arg? I see that the `.d.ts` generate just creates a `number` type.

Reply:
Great point! I tested this and the app does crash because the function is expecting an `int`. 
I tried guarding against it with CanGoToOffset but the Chromium implementation is the same as what's already checked in GetEntryAtIndex. Basically if the user puts a decimal in our current  `GoToOffset` or `GoToIndex` API, they would also crash. So I don't think there is anything we can do but let me know if you have any ideas!

```
content::NavigationEntry* WebContents::GetNavigationEntryAtIndex(
    int index) const {
  if (!CanGoToOffset(index))
    return nullptr;
  return web_contents()->GetController().GetEntryAtIndex(index);
}
```